Sources/NIO/SocketAddresses.swift

```diff
@@ -325,6 +327,34 @@ public enum SocketAddress: CustomStringConvertible {
             throw SocketAddressError.failedToParseIPString(ipAddress)
         }
     }
+    
+    /// Create a new `SocketAddress` for an IP address in ByteBuffer form.
+    ///
+    /// - parameters:
+    ///     - ipAddress: The IP address, in ByteBuffer form.
+    ///     - port: The target port.
+    /// - returns: the `SocketAddress` corresponding to this string and port combination.
+    /// - throws: may throw `SocketAddressError.failedToParseIPByteBuffer` if the IP address cannot be parsed or `SocketAddressError.unsupported` if the ATYP is wrong.
+    public init(ipAddress: ByteBuffer, port: Int) throws {
```

Comment:
One minor nit: can we change the argument to `packedIpAddress`? This will give us a slightly more descriptive name, and it'll avoid a type-based overload and replace it with a label-based overload.